README.md

```diff
@@ -30,59 +30,30 @@ number tag as well, e.g. `rust:1.45.1`. Similarly we could publish a new tag
 for each minor patch (`geo-ci:rust-1.50.0`, `geo-ci:rust-1.50.1`, etc.), but
 running CI against each patch seems like overkill at this point.
 
-## How to Update Rust
+**New images are built by Github Actions and published to the Github Container Registry if tests pass**
```

Comment:
The containers are published regardless of if the PR is merged, right? Ideally it'd be nice to not publish until merged into `main` but it seems non-trivial - building the three downstream containers across different jobs is nice for parallelism, but it means we can't rely on the local registry.

I can live with things as you have them now - but if you can think of an easy way to hold off on publishing until an merge into main, that'd be cool. Otherwise we can leave it for future work.

Reply:
GHA now has the ability to trigger a job if a PR is merged.
In order to take advantage of that, we'd have to add a few new moving parts:

1. Upload built `libproj-builder` x rust versions (for now) images as artifacts
2. Upload each sub-image x rust version artifact
3. On successful merge, download all the above again, then push to the registry.

It can be done, but it'll be easier if we can implement your no-Rust-for-main-image idea first. It will also mean duplicating the rust version and image name matrices because matrices can't be shared between jobs. That strikes me as error-prone, so we might have to explore building matrices from JSON files at job run time.

Reply:
My first though was to keep the publish step in the existing workflow, but gate it on a `if branch == main` check.

I'm happy to follow up with that later though.

Reply:
Oh, that might be a lot easier…